modules/islandora_core_feature/islandora_core_feature.install

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/**
+ * @file
+ * Update Hooks.
+ */
+
+use Drupal\field\Entity\FieldStorageConfig;
+
+/**
+ * Updates Media file size field storage for larger files.
+ */
+function islandora_core_feature_update_8001() {
+
+  $entity_type = 'media';
+  $field_name = 'field_file_size';
+
+  $database = \Drupal::database();
+  $data = [];
+  $storage_key = $entity_type . '.field_schema_data.' . $field_name;
+  $storage_schema = \Drupal::keyValue('entity.storage_schema.sql');
+  $field_schema = $storage_schema->get($storage_key);
+
+  foreach (["{$entity_type}__{$field_name}", "{$entity_type}_revision__{$field_name}"] as $table) {
+    // Squirrel away the data.
+    $data[$table] = $database->select($table, 'n')
+      ->fields('n')
+      ->execute()
+      ->fetchAll();
+
+    // Clean it out for resizing.
+    $database->truncate($table)->execute();
```

Comment:
I attempted to resize without cleaning it out and Drupal simply would not let me. Cleaning it out is the only way to save it. I suppose we could create a temporary table instead of relying on an in-memory store... 🤔

According to [the Drupal notes](https://www.drupal.org/node/2993033) this is how you truncate a table.

Reply:
Somewhat ambiguous due to nomenclature, but I'm assuming you don't mean a "temporary table" as supported by various DB engines which are automatically deleted at the end of the DB session, as they would exhibit the same issue of losing the stored data if something failed before getting it back in... but yeah, a short-lived table we create/check-for and populate, and delete when we're done... assuming success with in-place alteration with `changeField()` doesn't make it irrelevant.

Reply:
Looks like Drupal won't let changeField update the table without cleaning it out first either: "[error]  The SQL storage cannot change the schema for an existing field (field_file_size in media entity) with data." 

I'll see about creating a table to keep this all in, should the update fail and need to be attempted again.

Reply:
Instead of a new table I created a temporary file to stash a JSON copy of each table before truncating. I then re-read the file to insert the data back in after the schema is updated.

Reply:
Slight paranoia arising due to these files being `public://`-ly accessible, but _should_ be a fairly small window for things being exposed; though, I guess strictly that the amount of time would be related to the number of media in the table... actually, on that note... should this update hook be doing [the re-entrant batch thing](https://api.drupal.org/api/drupal/core!lib!Drupal!Core!Extension!module.api.php/function/hook_update_N/8.9.x#sec_bulk)?

There is something of a recipe to facilitate altering for widening changes, in: https://www.drupal.org/docs/drupal-apis/update-api/updating-database-schema-andor-data-in-drupal-8#s-altering-the-length-of-a-field-with-data

There, they're changing the length of a varchar (to a larger value, but not strongly enforced)... given we're wanting to make a similar more-or-less widening change of a datatype; though... I guess we're not strictly widening, since we're moving to exclude negative values... true, a negative file size may not make sense outside the realm of [magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming)), but... seems like an edge-case that could be easily covered, just by aborting the update and reporting that some manual intervention will be necessary before the update can run?

Reply:
As for using public, I didn't like the alternatives. Not all sites have a private filesystem set (a non-starter for an update hook) and temporary is at the whims of the OS' clean-up, so I'd rather not trust it for keeping our data dump in the case of a failure. At least a site admin _knows_ where public is if they have to go grab it should the site fail. Also, it is just a list of file sizes.... That said, if y'all prefer we use the temporary file-space, it is a simple change; just confirm either way.

Ah, update hook batches (are a pain to use properly). I recently fought with this on a separate internal migration.  Batch works best when the first and third steps are performed once and the middle step is repeated. (E.g. moving data from one field to another where the first phase sets up the new field, the middle step gradually copies, and the final step cleans up. Or a single repeatable step like modifying values in place.) Our three phases—dump all the data, make the change, re-import everything—are the inverse where the first and third can be done in chunks but the middle phase can't. Batching isn't suitable for this scenario.

I've attempted to use the strategy suggested by that recipe for in-place editing, but Drupal still balks at it.

Reply:
I should also note that the pain of creating new tables, copying the data in then out, then dropping them didn't seem worth it compared to doing the same with temporary files _briefly_ in the public filesystem. The attack window (knowing it exists and when) and reward of this is _really_ small.

Reply:
This stuck in my head last night; I might be able to use batch on this after all by nesting the batches within stages...

Reply:
Instead of dumping to files and what not, might it be simpler to:
* rename the target tables (assuming that renaming doesn't break similarly to the `::changeField()` business.
* create the new tables using the up-to-date schema
* copy table content using [insert/from](https://www.drupal.org/docs/drupal-apis/database-api/insert-queries#inserting-result-select), from the renamed to the newly created tables
* drop the renamed copies

... if not, couldn't whatever batch structure be accomplished with _multiple_ update hook implementations? A batch update hook followed by a non-batch, followed by a batch, kind of thing?

... any which way... still have the potential issue with the type change not being strictly widening with the move to exclude negative values... are we just ignoring this, or should we check for anything negative before trying to process, or... are we happy just letting it error how however it might?

Reply:
Oh, nice. I had seen references to Schema::copyTable which was no longer available. I hadn't see the rename table function.

As I eluded to in my previous comment, I think we can do the batching in a single hook. Also, with the table rename tactic, we only really need to batch the insert/from, which is manageable.

I do think we need to error, or at least warn, on negative file sizes. I mean, semantically speaking, those values are *already* broken.